src/OSPSuite.Presentation/UICommands/RunMultipleParameterIdentificationsUICommand.cs

```diff
@@ -0,0 +1,30 @@
+﻿using OSPSuite.Core.Domain.ParameterIdentifications;
+using OSPSuite.Core.Domain.Services.ParameterIdentifications;
+using OSPSuite.Core.Extensions;
+using OSPSuite.Core.Services;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OSPSuite.Presentation.UICommands
+{
+    public class RunMultipleParameterIdentificationsUICommand : ActiveObjectUICommand<IReadOnlyList<ParameterIdentification>>
+   {
+      private readonly IParameterIdentificationRunner _parameterIdentificationRunner;
+
+      public RunMultipleParameterIdentificationsUICommand(
+         IParameterIdentificationRunner parameterIdentificationRunner,
+         IActiveSubjectRetriever activeSubjectRetriever)
+         : base(activeSubjectRetriever)
+      {
+         this._parameterIdentificationRunner = parameterIdentificationRunner;
+      }
+
+      protected override async void PerformExecute()
+      {
+         var tasks = Subject.Select(pi => _parameterIdentificationRunner.SecureAwait(x => x.Run(pi)));
```

Comment:
@rwmcintosh SecureAWai hs going to await each of them sequentially. Is this the intent?

Reply:
That wasn't my intent, and I think it doesn't happen that way.  Although there is an await within the DoWithinExceptionHandler it is encapsulated in a task. each task is kicked off by _parameterIdentificationRunner.SecureAwait(x => x.Run(pi)). This part of the code doesn't wait for each task to complete, so that makes them running all concurrently. The await in 'DoWithinExceptionHandler' is just to properly catch errors in the exception handler there as far as I can see. 

I also tested it, and concluded that the two paramteridentifications runned concurrently. An earlier concept did not do that, so I saw the difference in behaviour.